### ISSUE TITLE
Customize preview HTML template

### ISSUE THREAD

**Issue by com2** [johanmattssonm/birdfont#75]
Is there a way to easily customize the HTML preview template, preferably without recompiling? I would like to make it more suitable for the creation of icon fonts.

**Comment by johanmattssonm** [comment]
Yes. You can edit the HTML file, it is located in the same folder as your exported TTF and OTF files.

**Comment by com2** [comment]
Yes, but when you frequently work with icon fonts you have to overwrite it every time. Would be nice when the software takes the HTML code from a template that can be customized by the user. For instance in a Linux version at installation (first time it runs) the software could write the template to .config/preview.html After that every time a font preview is generated, it will read .config/preview.html  and write it to myfontname.html. Similar scheme could work in other OS. The user then can make the change once in .config/preview.html and automatically it will be there in myfontname.html just like the user wants it. Just an idea to make it easier for the user.

**Comment by johanmattssonm** [comment]
Thank you for the suggestion. A preview template will be available in the next release.

**Comment by com2** [comment]
Cool! Will be donating soon!

**Comment by com2** [comment]
I am trying this feature with 3.15-free.flatpak and I either do not understand how this works or it is not yet working (like I expected). The template file is not found. To be sure I created a new .config/birdfont folder and created a new (empty) font with the "Private Use Area" character set. I then previewed it. By then I would have expected to appear a .config/birdfont/preview.html that I could edit. I need this example generated by BF because I will need to know the variable place holder for the font file name, that will have to change depending on the opened font project.

**Comment by johanmattssonm** [comment]
Please try to paste the config file in BirdFont settings. Here is the button:
![skarmbild fran 2018-11-19 14-38-54](https://user-images.githubusercontent.com/350392/48710630-44ec5580-ec09-11e8-8f35-f3bf00a8a8ef.png)

**Comment by com2** [comment]
That must be the paid version because I don't see that setting here. OK, I will donate and get it now.

**Comment by com2** [comment]
OK, I installed the birdfont-3.16.0-plus. I see it there. You want me to recheck the 3.15.0-free?

**Comment by johanmattssonm** [comment]
Thank you for you support. It would be nice to know it it isn't to much of problem.

**Comment by com2** [comment]
Yes, it works there too. I must have used an old version or used the flatpak tool wrongly. Thanks for your nice tool. Really worth it!

**Comment by com2** [comment]
I changed the template with the here indicated button to my liking (adding a number of "Private use area" characters in the preview with larger font-size). I expected that Preview exports would then show with my changes, but it didn't. That was because a preview html file pre-existed in the export folder and it will not get overwritten at Preview export. I discovered that the preview html only gets created once and is never overwritten by the software. That means that beside the template there is a second way to customise the preview html directly because it will never be changed by the software. This allows for per font customisations. When you delete the html file, BF will notice and write a new one according to template. Great job!
